aiverify-foundation/moonshot-ui
# Feature/use bookmark

## Description

This covers addition of a popup panel on red teaming session screen. It displays all the saved/bookmarked prompts. Selecting a bookmark and clicking `use` button, sends the saved prompt to the endpoints in the session

## Motivation and Context

Enable reusing of prompts

## Type of Change


 - feat: A new feature










## How to Test

1. Start a new or resume existing Red Teaming Session
2. Send prompts to endpoints
3. Save prompts as bookmarks (click on ribbon icon next to prompt in chat window)
4. Click on Bookmarked Prompts button on the left bottom panel
5. Verify the saved prompts on the popup
6. Select a prompt
7. Click Use button
8. Verify saved prompts are sent to endpoints in session

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x ]  I have tested the changes locally and verified that they work as expected.
- [ x]  I have added appropriate unit tests or functional tests for the changes made.
- [ x]  I have followed the project's coding conventions and style guidelines.
- [ x]  I have rebased my branch onto the latest commit of the main branch.
- [ x]  I have performed a self-review of my own code.
- [ x]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)



## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
